### DISCUSSION TITLE
Установка на Mac Os

### DISCUSSION THREAD

**Discussion by polinacheremisinova** [xvzc/SpoofDPI#75] [Q&A] [answered]
Автор, не очень понятно что конкретно нужно делать на мак ос, дабы заработало. Достаточно просто запустить файл с командой в терминале? Нужно ли что-то еще делать? Из инструкции не очень непонятно :( Разъясните пожалуйста!

**Comment by anrulev** [comment]


**Reply by polinacheremisinova** [reply]
Спасибо! А после этого по умолчанию будет активирован spoof dpi? Или надо каждый раз заново вводить команды в терминале. Извините за тупые вопросы, но я просто dumb в этих технических вопросах

**Reply by polinacheremisinova** [reply]
Спасибо большое за разъяснение и программу! У меня еще вопрос, что делать, если программа не дает работать другим программам на компе? В моем случае это программа Figma, при запущенном spoof программа перестает воспринимать соединение, и в терминале отображается ошибка "ERRO[2024-08-03T01:11:53+03:00] [PROXY] Error while dns lookup: www.figma.com couldn not resolve the domain". Подскажите пожалуйста!

**Reply by anrulev** [reply, marked as answer]
> Спасибо большое за разъяснение и программу! У меня еще вопрос, что делать, если программа не дает работать другим программам на компе? В моем случае это программа Figma, при запущенном spoof программа перестает воспринимать соединение, и в терминале отображается ошибка "ERRO[2024-08-03T01:11:53+03:00] [PROXY] Error while dns lookup: [www.figma.com](http://www.figma.com) couldn not resolve the domain". Подскажите пожалуйста!

закрыть окно терминала

**Reply by PoorMouse** [reply]
Тут автор вопроса благодарит за разъяснение, но я его нигде не вижу. Как же все-таки правильно пользоваться этой программой? Простого запуска в консоли недостаточно. Видимо, нужно делать дополнительные настройки.

**Reply by RexSatanas** [reply]
так после закрытия терминала процесс завершается. Разве нет

**Reply by lovipomidorku** [reply]
> так после закрытия терминала процесс завершается. Разве нет

Да, завершается, Терминал нужно держать открытым, если мешается, можно свернуть в док

**Reply by polinacheremisinova** [reply]
> Тут автор вопроса благодарит за разъяснение, но я его нигде не вижу. Как же все-таки правильно пользоваться этой программой? Простого запуска в консоли недостаточно. Видимо, нужно делать дополнительные настройки.

Просто нужно после установки запустить команду export PATH=$PATH:~/.spoof-dpi/bin
spoof-dpi , и держать терминал открытым, пока тебе нужно действие программы. Все. Доп. настроек не надо.

**Reply by iamrealmarsel** [reply]
> Просто нужно после установки запустить команду export PATH=$PATH:~/.spoof-dpi/bin spoof-dpi , и держать терминал открытым, пока тебе нужно действие программы. Все. Доп. настроек не надо.

У меня почему-то скрипт запускается без всяких ошибок, но Ютуб все равно не грузит. Можешь сказать какая версия макОса у тебя? Мне кажется с этим связано.

**Reply by polinacheremisinova** [reply]
> У меня почему-то скрипт запускается без всяких ошибок, но Ютуб все равно не грузит. Можешь сказать какая версия макОса у тебя? Мне кажется с этим связано.

У меня тоже периодически ютуб не работает, но я не думаю что это связано как-то со скриптом :( Вообще в разное время суток ютуб по разному отвечает, и скрипт то работает, то нет

**Reply by iamrealmarsel** [reply]
> 
> У меня тоже периодически ютуб не работает, но я не думаю что это связано как-то со скриптом :( Вообще в разное время суток ютуб по разному отвечает, и скрипт то работает, то нет

Хм, странно, сегодня заработало, у меня просто старый макбук с версией ос 10.15, думал в этом дело, но возможно в чем-то другом..

**Reply by gray1717** [reply]
> > так после закрытия терминала процесс завершается. Разве нет
> 
> Да, завершается, Терминал нужно держать открытым, если мешается, можно свернуть в док

а есть ли что-то на подобии ````dtach```` в linux что бы не держать терминал открытым?

**Comment by frerrr** [comment]
Please tell me how to uninstall on mac os?

**Reply by Wlatt** [reply]
rm -rf ~/.spoof-dpi

**Reply by frerrr** [reply]
thanks

**Comment by turbodrxg** [comment]
Хотел написать небольшую программу скрипт для запуска этого ПО, выдает ошибку - Ошибка: time="2024-08-07T02:12:59+03:00" level=fatal msg="[PROXY] Error creating listener: listen tcp4 127.0.0.1:8080: bind: address already in use"

Сам код такой - 
`import subprocess
import os
import tkinter as tk
from tkinter import messagebox
from tkinter import PhotoImage

# Функция для выполнения команды в терминале
def run_command(command):
    result = subprocess.run(command, shell=True, text=True, capture_output=True)
    if result.returncode == 0:
        return result.stdout
    else:
        return f"Ошибка: {result.stderr}"

# Функция для установки и запуска SpoofDPI
def install_spoofdpi():
    install_button.config(state="disabled")  # Деактивировать кнопку во время выполнения

    try:
        # Скачивание и установка SpoofDPI
        install_output = run_command("curl -fsSL curl -fsSL https://raw.githubusercontent.com/xvzc/SpoofDPI/main/install.sh | bash -s darwin-arm64")
        messagebox.showinfo("Установка SpoofDPI", install_output)
        
        # Добавление SpoofDPI в PATH
        spoof_dpi_bin_path = os.path.expanduser("~/.spoof-dpi/bin")
        os.environ["PATH"] += os.pathsep + spoof_dpi_bin_path
        messagebox.showinfo("Обновление PATH", f"SpoofDPI добавлен в PATH: {spoof_dpi_bin_path}")
        
        # Запуск spoof-dpi
        run_output = run_command("spoof-dpi")
        messagebox.showinfo("Запуск spoof-dpi", run_output)
    finally:
        install_button.config(state="normal")  # Активировать кнопку после выполнения

# Создание основного окна
root = tk.Tk()
root.title("Установка и запуск SpoofDPI")
root.geometry("400x300")
root.configure(bg="lightblue")

# Загрузка изображения для фона
try:
    bg_image = PhotoImage(file="background.png")
    bg_label = tk.Label(root, image=bg_image)
    bg_label.place(relwidth=1, relheight=1)
except Exception as e:
    print(f"Не удалось загрузить изображение: {e}")

# Кастомные стили
custom_font = ("Helvetica", 12, "bold")
button_style = {"bg": "purple", "fg": "white", "font": custom_font, "relief": "raised", "bd": 3}

# Создание кнопки для установки и запуска SpoofDPI
install_button = tk.Button(root, text="Установить и запустить SpoofDPI", command=install_spoofdpi, **button_style)
install_button.pack(pady=20)

# Запуск главного цикла приложения
root.mainloop()
`

**Reply by PoorMouse** [reply]
Эта ошибка, собственно, означает, что сочетание адреса и порта уже чем-то используется. Возможно, ранее запущенным процессом SpoofDPI. Попробуйте выбрать другой порт.

**Reply by algennadevich** [reply]
А как это сделать?

**Reply by PoorMouse** [reply]
При запуске приложения можно указать любой порт.

**Reply by algennadevich** [reply]
При запуске какого приложения?

**Reply by algennadevich** [reply]
Это же делается в терминале.

**Reply by lovipomidorku** [reply]
Я написал zsh скрипт для запуска. Может пригодится https://github.com/xvzc/SpoofDPI/discussions/115

**Comment by algennadevich** [comment]
Добрый день! После ввода команды SpoofDPI. Выдаёт такую строку:

FATA[2024-08-07T11:50:38+05:00] Error while changing proxy settings

Что надо сделать, чтобы заработало?

**Reply by algennadevich** [reply]
<img width="948" alt="Снимок экрана 2024-08-07 в 12 12 15 PM" src="https://github.com/user-attachments/assets/beed1e79-6aaa-424f-8a84-53b6d0aa8a06">
Вот скриншот терминала.

**Reply by alex-v-fraser** [reply]
`sudo spoof-dpi` пробовал?

**Reply by algennadevich** [reply]
> `sudo spoof-dpi` пробовал?

Нет. Он пароль просит какой-то?

**Reply by algennadevich** [reply]
И никак не реагирует на нажатие кнопок.

**Reply by PoorMouse** [reply]
Просит пароль администратора системы. Вы же sudo используете.

**Reply by ivansokolov45** [reply]
Такая же проблема с sudo spoof-dpi. Пробовал задавать другие порты, тключал vpn расширения в хроме.

**Reply by alexeymedvedev1** [reply]
Приветствую всех. 
Подскажите если найдете решение ошибки Error while changing proxy settings. 
через sudo и пароль (у кого не находит команду нужно вводить /usr/bin/sudo) выводит такую же ошибку.

**Reply by PoorMouse** [reply]
Посмотрите, нет ли у вас других приложений, работающих с прокси.

**Reply by alexeymedvedev1** [reply]
Спасибо, была такая мысль. Это ноут у друга, поэтому все возможно) 
До обновления на 15 версию, все работало.

**Reply by alexeymedvedev1** [reply]
@PoorMouse А подскажите, пожалуйста. Как лог включить / смотреть?

**Reply by PoorMouse** [reply]
Запустить программу с флагом -debug

**Reply by alexeymedvedev1** [reply]
Решил вопрос) 
Оказалось в команде Export PATH=$PATH:~/.spoof-dpi/bin был пропущен знак $. после чего сбивались все переменные окружения zsh и переставали работать даже стандарные команды ls, rm, sudo). 

Смог увидеть только когда сбросил zsh и запустил spoof-dpi из директории - и все заработало

Вдруг кому полезно будет  - сброс zsh делается через команду - exec zsh

Спасибо!
